### PR TITLE
Charles/sample dropdown

### PR DESCRIPTION
**Pull request by cdebourcy**
# Description
Allows you to navigate to another sample in the same project from the report page.

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. Go to report page
2. Click on sample name
3. Select another sample in the dropdown
4. Verify you're taken to the selected sample

## Impacted Areas in Application
- [X] Report Page

# Checklist:
- [X] I have run through the testing script to make sure current functionality is unchanged
- [X] I have done relevant tests that prove my fix is effective or that my feature works
- [X] I have spent time testing out edge cases for my feature

### CODE REVIEW

File: app/assets/javascripts/components/PipelineSampleReads.jsx [hunk 1]
```diff
@@ -351,6 +358,25 @@ class PipelineSampleReads extends React.Component {
...
     }
+    sample_dropdown = (
+      <div className='dropdown-button sample-select-dropdown' data-activates='sample-list'>
+        { this.sampleInfo.name }
+        <i className="fa fa-chevron-down right"/>
+      </div>
+    )
+    sample_dropdown_content = (
+      <ul id='sample-list' className='dropdown-content sample-dropdown-content'>
+       { Object.keys(this.sample_map).map((sample_id, i) => {
```
Comment (cyrielo) [inline app/assets/javascripts/components/PipelineSampleReads.jsx]: looks like sample_map is undefined, it breaks at this point

Comment (cdebourcy) [reply app/assets/javascripts/components/PipelineSampleReads.jsx]: It is defined in line 10 above. Can you give me an example of a URL where it breaks?

Comment (cyrielo) [reply app/assets/javascripts/components/PipelineSampleReads.jsx]: weird just checked out the branch again, couldn't reproduce this error, works fine.

Comment (cdebourcy) [reply app/assets/javascripts/components/PipelineSampleReads.jsx]: Weitd, maybe a github synch issue. Good point though about handling edge cases -- I've added some checks to omit the dropdown if there is a problem with sample_map or if it contains only a single sample

**Comment by cyrielo** [comment]
That make sens

**Pull request merged**